### PR TITLE
fix(log): fix bug that logs were not cleaned in time.

### PR DESCRIPTION
**Pull request by shuqiang-zheng**
close: #2399
**What this PR does / why we need it**:
fix bug that logs were not cleaned in time.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: util/log/rotate.go [hunk 1]
```diff
@@ -22,7 +22,8 @@ const (
 	// DefaultHeadRoom The tolerance for the log space limit (in megabytes)
 	DefaultHeadRoom = 50 * 1024
 	// DefaultHeadRatio The disk reserve space ratio
-	DefaultHeadRatio = 0.2
+	DefaultHeadRatio         = 0.2
+	DefaultLogLeftSpaceLimit = 1 * 1024
```
Comment (leonrayang) [inline util/log/rotate.go:26]: 5G

File: util/log/log_test.go [hunk 2]
```diff
@@ -78,6 +79,69 @@ func TestLog(t *testing.T) {
...
+	time.Sleep(200 * time.Millisecond)
+
+	_, err = os.Stat(logFilePath1)
+	if os.IsNotExist(err) {
+		t.Errorf("expect file[%v] doesn't exist but err is [%v]", logFilePath1, err)
+		return
+	}
+}
+
+func TestLogLeftSpaceLimit02(t *testing.T) {
```
Comment (leonrayang) [inline util/log/log_test.go:114]: duplicated code too much

File: util/log/log.go [hunk 3]
```diff
@@ -309,6 +309,9 @@ func InitLog(dir, module string, level Level, rotate *LogRotate) (*Log, error) {
 		} else {
 			minRatio = float64(fs.Blocks*uint64(fs.Bsize)) * DefaultHeadRatio / 1024 / 1024
 		}
+
+		minRatio = math.Max(minRatio, float64(logLeftSpaceLimit))
```
Comment (leonrayang) [inline util/log/log.go:313]: minRatio name not meaningful

**Review by leonrayang — DISMISSED** [review]
LGTM

**Review by leonrayang — APPROVED** [review]
LGTM

**Review by Victor1319 — APPROVED** [review]
LGTM

**Pull request merged**